Add a lighter name mangling option (#14695)
coq/coq: doc/changelog/07-vernac-commands-and-options/14695-alizter+mangle-names-light.rst

- **Added:**

>   New :flag:`Mangle Names Light` flag. When set will change the behaviour of :flag:`Mangle Names`. For example, the name `foo` becomes `_0` with :flag:`Mangle Names`, but with :flag:`Mangle Names Light` set, will become `_foo`.

Reviewer: ```suggestion
  The :flag:`Mangle Names Light` flag, which changes the behavior of :flag:`Mangle Names`. For example, the
  name `foo` becomes `_0` with :flag:`Mangle Names`, but with :flag:`Mangle Names Light` set, will become `_foo`
```
"New" is redundant since it's Added.
We prefer US spellings in the doc.
And we like to keep lines a reasonable length (I usually start a new line after 80 characters or so.)

Author: Ah I wasn't sure if I was supposed to hardwrap the lines in rst documents. No problem

Reviewer: It seems to be the convention.  Same thing for the code.  Note that `.. tacn:`, `.. exn::` can't be wrapped.
